### PR TITLE
add: progress bar when running with optimizations all

### LINKED ISSUE #3124: Add progress bars to optimization mode

### PR DESCRIPTION
**Pull request by brendongo**
Closes:  https://github.com/returntocorp/semgrep/issues/3124

### CODE REVIEW

File: semgrep/semgrep/core_runner.py [hunk 1]
```diff
@@ -613,90 +613,86 @@ def _run_rules_direct_to_semgrep_core(
...
+                    findings = create_output(rule, pattern_matches)
+
+                    # TODO: we should do that in Semgrep_generic.ml instead
+                    findings = dedup_output(findings)
+                    outputs[rule].extend(findings)
+                    errors.extend(
+                        CoreException.from_json(
+                            e, language.value, rule.id
+                        ).into_semgrep_error()
+                        for e in output_json["errors"]
```
Comment (brendongo) [inline semgrep/semgrep/core_runner.py:692]: This change is just indenting the code

File: semgrep/semgrep/core_runner.py [hunk 2]
```diff
@@ -613,90 +613,86 @@ def _run_rules_direct_to_semgrep_core(
...
-                        "-target_file",
-                        target_file.name,
-                        "-use_parsing_cache",
-                        semgrep_core_ast_cache_dir,
-                        "-timeout",
-                        str(self._timeout),
-                        "-max_memory",
-                        str(self._max_memory),
-                        "-json_time",
+                for language in rule.languages:
```
Comment (brendongo) [inline semgrep/semgrep/core_runner.py:627]: This is the only relevant change. The rest of the diff below is indentation

File: semgrep/semgrep/core_runner.py [hunk 3]
```diff
@@ -613,90 +613,86 @@ def _run_rules_direct_to_semgrep_core(
...
-            for rule, language in tuple(
-                chain(
-                    *(
-                        [(rule, language) for language in rule.languages]
-                        for rule in rules
-                    )
-                )
+            for rule in progress_bar(
+                rules, bar_format="{l_bar}{bar}|{n_fmt}/{total_fmt}"
             ):
```
Comment (brendongo) [inline semgrep/semgrep/core_runner.py:626]: Changing from chain in for loop to double for loop is the only relevant diff in this PR. The rest are indentation changes.

Comment (aryx) [reply semgrep/semgrep/core_runner.py:626]: github needs to improve his diff algorithm to show those just-change-of-indentation lines in a different color.

**Review by aryx — APPROVED** [review]

**Pull request merged**